mono/metadata/class-init.c

```diff
@@ -1260,6 +1260,11 @@ make_generic_param_class (MonoGenericParam *param)
 	/* We don't use type_token for VAR since only classes can use it (not arrays, pointer, VARs, etc) */
 	klass->sizes.generic_param_token = !is_anonymous ? pinfo->token : 0;
 
+	if (param->gshared_constraint) {
+		MonoClass *constraint_class = mono_class_from_mono_type_internal (param->gshared_constraint);
+		mono_class_init_sizes (constraint_class);
```

Comment:
I don't think there's a problem, but I'm curious if there's a concise explanation why this can't get into a loop.  there's no weird example where the constraint derived from something that depends on `klass` already existing?

Reply:
The gparams which have gshared_constraint set are created by mini_get_shared_gparam () in mini-generic-sharing.c from an already existing constraint.